test/e2e/l7networkpolicy_test.go

```diff
@@ -289,6 +293,77 @@ func testL7NetworkPolicyHTTP(t *testing.T, data *TestData) {
+	_, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace)
+	require.NoError(t, err, "Expected IP for Pod '%s'", clientPodName)
+
+	// Create a hostNetwork server Pod on another Node, with the Pod being selected by the test NetworkPolicy as the
+	// destination.
+	serverPodName := "test-l7-http-server"
+	serverPodLabels := map[string]string{"test-l7-http-e2e": "server"}
+	cmd := []string{"/agnhost", "netexec", "--http-port=8081"}
+	require.NoError(t, NewPodBuilder(serverPodName, data.testNamespace, agnhostImage).
+		WithHostNetwork(true).
+		OnNode(nodeName(1)).
+		WithCommand(cmd).
+		WithLabels(serverPodLabels).
+		Create(data))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, serverPodName, data.testNamespace)
+	require.NoError(t, err, "Expected IP for Pod '%s'", serverPodName)
+	serverIPs := podIPs.AsSlice()
+
+	l7ProtocolAllowsPathShell := []crdv1beta1.L7Protocol{
+		{
+			HTTP: &crdv1beta1.HTTPProtocol{
+				Method: "GET",
+				Path:   "/shell*",
+			},
+		},
+	}
+	policyAllowPathShellName := "test-l7-http-allow-path-shell"
+	createL7NetworkPolicy(t, data, true, policyAllowPathShellName, 1, clientPodLabels, serverPodLabels, ProtocolTCP, p8081, l7ProtocolAllowsPathShell)
+	time.Sleep(networkPolicyDelay)
+
+	for _, ip := range serverIPs {
+		baseURL := net.JoinHostPort(ip.String(), strconv.Itoa(int(p8081)))
+		// Verify that the test L7 NetworkPolicy allows access to the "/shell" path with large body payload.
+		assert.Eventually(t, func() bool {
+			// Get the MTU of the test client Pod, assuming it's the MTU of the K8s cluster.
+			cmd := []string{"cat", "/sys/class/net/eth0/mtu"}
+			mtuStdout, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, agnhostContainerName, cmd)
+			if err != nil {
+				return false
+			}
+			mtu, err := strconv.Atoi(strings.TrimSpace(mtuStdout))
+			if err != nil {
+				return false
+			}
+
+			// Run the command that makes the test server send an HTTP response with a body larger than the MTU on the
+			// test client Pod.
+			testBodySize := mtu * 2
+			cmd = []string{"curl", "--data-urlencode", fmt.Sprintf(`cmd=head -c %d </dev/zero | tr '\0' 'A'`, testBodySize), fmt.Sprintf("http://%s/shell?cmd", baseURL)}
+			stdout, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, agnhostContainerName, cmd)
+			if err != nil || !strings.Contains(stdout, strings.Repeat("A", testBodySize)) {
+				return false
+			}
+			return true
+		}, 5*time.Second, time.Second)
+
+		// Verify that the test L7 NetworkPolicy denies access to the "/hostname" path.
+		assert.Eventually(t, func() bool {
+			cmd := []string{"wget", "-O", "-", fmt.Sprintf("%s/%s", baseURL, "hostname"), "-T", "1", "-t", "1"}
+			_, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, agnhostContainerName, cmd)
+			return err != nil
+		}, 5*time.Second, time.Second)
```

Comment:
it may be more meaningful to have this assertion first
this is because once we observe `/hostname` being denied, it's fair to assume that the policy has been realized. Until the policy has been realized, access to `/shell` will be allowed regardless, so the previous assertion may succeed even if the policy has not been realized yet.

Reply:
Good catch! It should be tested first to verify the test policy!